### PR TITLE
#107 - extend Year service and repository to return year by its tenant ID

### PR DESCRIPTION
**Pull request by gregito**
also, updating the `MultipleElementError` to implement the `Exception` instead of `Error`, since Errors should be restricted for application-wide error events when further operations are not possible

### CODE REVIEW

**Review by lmarti95 — APPROVED** [review]

File: lib/data/repositories/year_repository.dart [hunk 1]
```diff
@@ -54,7 +54,27 @@ class YearRepository extends CrudRepository<Year> {
...
+
+  Future<Year?> getByTenant(String tenantId) async {
+    QuerySnapshot<Map<String, dynamic>> yearsSnapshot = await firestore
+        .collection('years')
+        .where('tenantId', isEqualTo: tenantId)
+        .get();
+    if (yearsSnapshot.size > 0) {
+      List<Year> yearList = yearsSnapshot.docs.map((yearDoc) {
+        Map<String, dynamic> data = yearDoc.data();
+        data.putIfAbsent('id', () => yearDoc.id);
```
Comment (terjekig) [inline lib/data/repositories/year_repository.dart:72]: There is no "id" field in "years", so this would always put documentID

**Review by terjekig — CHANGES_REQUESTED** [review]
1 small thing

File: lib/data/repositories/year_repository.dart (hunk 1 above)
Comment (gregito) [reply lib/data/repositories/year_repository.dart:72]: exactly, but the Map type doesn't have a simple `put` method the append an extra key-value pair into it, thus this has to happen to be able to create a proper Year object

Comment (terjekig) [reply lib/data/repositories/year_repository.dart:72]: Ty, I misunderstood the aim of this.

**Review by terjekig — APPROVED** [review]

**Pull request merged**